settingsmetadata/repository/tenantSettingsRepository.go

```diff
@@ -0,0 +1,102 @@
+package repository
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/islax/microapp/config"
+	"github.com/islax/microapp/repository"
+	"github.com/islax/microapp/settingsmetadata/model"
+	uuid "github.com/satori/go.uuid"
+)
+
+//TenantSettingsRepository
+type TenantSettingsRepository interface {
+	repository.Repository
+	GetTenantSettings(uow *repository.UnitOfWork, tenantID uuid.UUID) (map[string]string, error)
+}
+
+//NewAlertRepository
+func NewTenantSettingsRepository(config *config.Config) TenantSettingsRepository {
+	return &gormTenantSettingsRepository{Config: config}
+}
+
+type gormTenantSettingsRepository struct {
+	repository.GormRepository
+	settingsMetadatas       []model.SettingsMetaData
+	globalsettingsMetadatas []model.SettingsMetaData
+	*config.Config
+}
+
+func (tenantRepository *gormTenantSettingsRepository) GetTenantSettings(uow *repository.UnitOfWork, tenantID uuid.UUID) (map[string]string, error) {
+	tenant := model.TenantSettings{}
+	tenant.ID = tenantID
+	queryProcessor := []repository.QueryProcessor{repository.Filter("id = ?", tenantID)}
+	if err := tenantRepository.GetFirst(uow, &tenant, queryProcessor); err != nil {
+		if !err.IsRecordNotFoundError() {
+			return nil, err
+		}
+	}
+
+	if err := tenantRepository.checkAndInitializeSettingsMetadata(); err != nil {
+		return nil, err
+	}
+
+	settingsmetadata := tenantRepository.settingsMetadatas
+	if tenantID.String() == "00000000-0000-0000-0000-000000000000" {
+		settingsmetadata = tenantRepository.globalsettingsMetadatas
+	}
+
+	err := tenant.GetTenantSettings(settingsmetadata)
+	if err != nil {
+		return nil, err
+	}
+
+	settingsMap, err := tenant.GetSettings()
+	if err != nil {
+		return nil, err
+	}
+
+	//convert to map[string]string
+	returnMap := make(map[string]string)
+	for key, setting := range settingsMap {
+		returnMap[key] = fmt.Sprintf("%v", setting)
+	}
+
+	return returnMap, nil
+}
+
+func (tenantRepository *gormTenantSettingsRepository) checkAndInitializeSettingsMetadata() error {
+	if len(tenantRepository.settingsMetadatas) == 0 {
+		settingMetadata, err := tenantRepository.initSettingsMetaData(config.EvSuffixForSettingsMetadataPath)
+		if err != nil {
+			return err
+		}
+		tenantRepository.settingsMetadatas = settingMetadata
+	}
+	if len(tenantRepository.globalsettingsMetadatas) == 0 {
```

Comment:
controller.app.Config.IsSet(config.EvSuffixForGlobalSettingsMetadataPath) is not present here the way it has been done in settingsmetadata controller

Reply:
condition added